src/ToricVarieties/NormalToricVarieties/constructors.jl

```diff
@@ -479,95 +479,73 @@ function Base.:*(v::AbstractNormalToricVariety, w::AbstractNormalToricVariety)
 end
 
 
-############################
-### 5: Display
-############################
-function Base.show(io::IO, v::AbstractNormalToricVariety)
-    # initiate properties string
-    properties_string = ["A normal"]
-
-    # affine?
-    if has_attribute(v, :isaffine)
-        if get_attribute(v, :isaffine)
-            push!(properties_string, "affine")
-        else
-            push!(properties_string, "non-affine")
-        end
-    end
-    
-    # smooth/simplicial?
-    if has_attribute(v, :issmooth)
-        if get_attribute(v, :issmooth)
-            push!(properties_string, "smooth")
+# If the property is A, then the callback can be used to ask for property B
+# with A=>B. Then if A is true, only A is displayed, and if B is false, only B
+# is displayed.
+function push_attribute_if_exists!(result::Vector{String}, 
+        v::T,
+        property::Symbol, name::String, 
+        alt_string::Union{String,Nothing}=nothing; 
```

Comment:
Could one also do this and then simplify the code accordingly? I can't try it myself right now, so it might be nonsense, but I thought this kind of code supported by Julia? Anyway, unimportant overall :-)
```suggestion
        alt_string::String="non-"*name; 
```

Reply:
Thanks, that works and even removes a few lines. :)